### PR TITLE
Fix the bug when there is an executable path including a space in windows

### LINKED ISSUE #721: Runtime Error occurs in windows runtime distribution when "-e" argument is provided.
### Description:
When gateway starts with "-e" parameter the following runtime error occurs.

>ballerina: unknown option: -e abc=abc C:\Users\wso2\Documents\product-microgateway\distribution\target\wso2am-micro-gw-windows-3.0.2-SNAPSHOT\wso2am-micro-gw-windows-3.0.2-SNAPSHOT\bin\EndpointReferenceProject.balx

### Steps to reproduce:
Try to use overriding backend endpoints as mentioned in [readme.md](https://github.com/wso2/product-microgateway/tree/v3.0.1#6-override-backend-service-connection-urls).

### Affected Product Version:
3.0.1

### OS, DB, other environment details and versions:
- OS: Windows 10

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
To resolve the issue when a parameter is provided using "-e" during the gateway start.
In addition, the executable path should be provided with double-quotes. 
>ex: gateway "pathToExecutable"

### Issues
Fixes #721

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments
Windows 10

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: distribution/resources/bin/gateway.bat [hunk 1]
```diff
@@ -62,15 +62,43 @@ if not exist %GW_HOME%\runtime\ (
...
+	    set "BAL_ARGS=%BAL_ARGS% \"%~1\""
+        goto :callBallerina
+	) else (
+            if "%~1"=="--debug" (
+                set "BAL_ARGS=%BAL_ARGS% %1 %2"
+                shift
+                shift
+                goto :formatAndValidateCmdArgs
+            ) else (
+				echo %~1
```
Comment (praminda) [inline distribution/resources/bin/gateway.bat]:
```suggestion
                echo %~1
```

File: distribution/resources/bin/gateway.bat [hunk 2]
```diff
@@ -62,15 +62,43 @@ if not exist %GW_HOME%\runtime\ (
     )
 )
 
+REM Needs to identify the ballerina arguments and the last argument which is the path of executable.
+REM The path of executable should be provided as \"<path>\" to avoid ballerina when the path includes a space.
+REM BAL_ARGS variable is used to store formatted string
+set BAL_ARGS=
+:formatAndValidateCmdArgs
+if "%~1"=="-e" (
```
Comment (praminda) [inline distribution/resources/bin/gateway.bat]: Indentation need to be corrected for the label content

File: distribution/resources/bin/gateway.bat (hunk 1 above)
Comment (VirajSalaka) [reply distribution/resources/bin/gateway.bat]: resolved with https://github.com/wso2/product-microgateway/pull/725/commits/a1bb55161bc993a4b5ea01805ca049af949a85fe

File: distribution/resources/bin/gateway.bat (hunk 2 above)
Comment (VirajSalaka) [reply distribution/resources/bin/gateway.bat]: resolved with https://github.com/wso2/product-microgateway/pull/725/commits/a1bb55161bc993a4b5ea01805ca049af949a85fe

**Pull request merged**